plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -18,217 +16,290 @@ import (
+	connection := plugin.Connect()
+	require.Error(t, connection)
+	require.Equal(t, "MSI not available", connection.Error())
+}
 
-				expectedTags := tC.expected["tags"].(map[string]interface{})
-				require.Equal(t, expectedTags, createdFakeIngestor.actualOutputMetric["tags"])
+func TestInit(t *testing.T) {
+	t.Parallel()
+	fakeClient := kusto.NewMockClient()
+	plugin := AzureDataExplorer{
+		Log:       testutil.Logger{},
+		Endpoint:  "someendpoint",
+		Database:  "databasename",
+		client:    fakeClient,
+		ingestors: make(map[string]ingest.Ingestor),
+	}
+	initResponse := plugin.Init()
+	require.Equal(t, initResponse, nil)
+}
 
-				expectedTime := tC.expected["timestamp"].(float64)
-				require.Equal(t, expectedTime, createdFakeIngestor.actualOutputMetric["timestamp"])
+func TestCreateRealIngestorManaged(t *testing.T) {
+	t.Parallel()
+	kustoLocalClient := kusto.NewMockClient()
+	localIngestor, err := createIngestorByTable(kustoLocalClient, "telegrafdb", "metrics", "managed")
+	require.Nil(t, err)
+	require.NotNil(t, localIngestor)
+}
 
-				if tC.createTables {
-					createTableString := fmt.Sprintf(createTableCommandExpected, expectedNameOfTable)
-					require.Equal(t, createTableString, tC.client.queries[0])
+func TestCreateRealIngestorQueued(t *testing.T) {
+	t.Parallel()
+	kustoLocalClient := kusto.NewMockClient()
+	localIngestor, err := createIngestorByTable(kustoLocalClient, "telegrafdb", "metrics", "queued")
+	require.Nil(t, err)
+	require.NotNil(t, localIngestor)
+}
 
-					createTableMappingString := fmt.Sprintf(createTableMappingCommandExpected, expectedNameOfTable, expectedNameOfTable)
-					require.Equal(t, createTableMappingString, tC.client.queries[1])
-				} else {
-					require.Empty(t, tC.client.queries)
-				}
-			}
-		})
-	}
+func TestInvalidIngestorType(t *testing.T) {
+	t.Parallel()
+	kustoLocalClient := kusto.NewMockClient()
+	localIngestor, err := createIngestorByTable(kustoLocalClient, "telegrafdb", "metrics", "streaming")
+	require.NotNil(t, err)
+	require.Nil(t, localIngestor)
+	require.Equal(t, "ingestion_type has to be one of managed or queued", err.Error())
 }
 
-func TestInitBlankEndpoint(t *testing.T) {
-	plugin := AzureDataExplorer{
-		Log:            testutil.Logger{},
-		client:         &fakeClient{},
-		ingesters:      map[string]localIngestor{},
-		createIngestor: createFakeIngestor,
+func TestClose(t *testing.T) {
+	t.Parallel()
+	fakeClient := kusto.NewMockClient()
+	adx := AzureDataExplorer{
+		Log:       testutil.Logger{},
+		Endpoint:  "someendpoint",
+		Database:  "databasename",
+		client:    fakeClient,
+		ingestors: make(map[string]ingest.Ingestor),
 	}
+	err := adx.Close()
+	require.Nil(t, err)
+	// client becomes nil in the end
+	require.Nil(t, adx.client)
+	require.Nil(t, adx.ingestors)
+}
 
-	errorInit := plugin.Init()
-	require.Error(t, errorInit)
-	require.Equal(t, "Endpoint configuration cannot be empty", errorInit.Error())
+type mockIngestor struct {
+	records []string
 }
```

Comment:
I'm sorry I may be missing the point here. The mockIngestor is something we created newly to accommodate the test cases. 
Please correct me if the comment meant something else.

Reply:
Well but you removed `type fakeIngestor struct` didn't you? Why? Couldn't you just modify the previously existing struct?

Reply:
Thanks for the comment @srebhan. 

In ADX there is the `client` and `ingestor`. The client is responsible for the queries and management APIs of the cluster and `ingestor` is responsible for ingesting the data into the DB tables. In the previous version we have created our own fake client for the testing purpose, and this has to support 2 attributes `queries` and `internalMgmt`. In the newer GO SDKs for versions for ADX, this is replaced by the core SDKs mockClient (`kusto.NewMockClient()` ) hence the use of fakeClient was redundant. 

Since the ingestor is instantiated from the client the attributes and the behaviors also changed and hence the struct  `mockIngestor` is renamed (to be consistent with `mockClient` -> `mockIngestor` as opposed to `fakeClient` -> `fakeIngestor`). 

if it is mandatory, we can still go for rollback of the name of the struct.

Reply:
No leave it as is...